k8s_test_harness/util/env_util.py

```diff
@@ -0,0 +1,204 @@
+#
+# Copyright 2024 Canonical, Ltd.
+# See LICENSE file for licensing details
+#
+
+"""Module containing utilities for extracting env options used in testing.
+
+This module is intended to be used with pre-built ROCK meta information
+produced by the canonical/k8s-workflows/build_rocks.yaml workflow:
+
+# TODO(aznashwan): pin exact link upon first stable release of workflows:
+https://github.com/canonical/k8s-workflows/blob/main/.github/workflows/build_rocks.yaml#L125-L133
+"""
+
+import json
+import os
+from dataclasses import dataclass
+from typing import Dict, List
+
+
+DEFAULT_BUILT_ROCKS_METADATA_ENV_VAR = "BUILT_ROCKS_METADATA"
+
+
+def must_get_env_var(var_name: str) -> str:
+    """Reads the environment variable with the given name.
+
+    :raises EnvironmentError: when the variable isn't set or is empty.
+    """
+    val = os.getenv(var_name, default='')
+    if not val:
+        raise EnvironmentError(
+            f"Environment variable '{var_name}' is not defined. "
+            f"Current environment contains: {dict(os.environ)}")
+    return val
+
+
+@dataclass
+class RockMetaInfo:
+    """Class representing metadata for a built ROCK as produced by the
+    canonical/k8s-workflows/build_rocks.yaml workflow:
+
+    # TODO(aznashwan): pin exact link upon first stable release of workflows:
+    https://github.com/canonical/k8s-workflows/blob/main/.github/workflows/build_rocks.yaml#L125-L133
```

Comment:
Just a reminder comment to update before merging this

Reply:
This requires a tag/release on the workflows (which is why the URL is currently linking to the `main` branch).

Without it, I could either link the current HEAD of the repo or remove the comment entirely.